elixir/apps/domain/priv/repo/migrations/20240618210715_add_created_by_actor.exs

```diff
@@ -0,0 +1,37 @@
+defmodule Domain.Repo.Migrations.AddCreatedByActor do
+  use Ecto.Migration
+
+  @table_names ~w[
+    actor_groups
+    auth_identities
+    auth_providers
+    gateway_groups
+    policies
+    relay_groups
+    resources
+    resource_connections
+    tokens
+  ]a
+
+  defp migrate_data(table_name) do
+    """
+    UPDATE #{table_name} AS t
+    SET created_by_actor_id = ai.actor_id
+    FROM auth_identities AS ai
+    WHERE t.created_by_identity_id = ai.id
+    AND t.created_by = 'identity'
+    AND t.created_by_identity_id is not null;
+    """
+    |> execute("")
+  end
+
+  def change do
+    for table_name <- @table_names do
+      alter table(table_name) do
+        add(:created_by_actor_id, :uuid)
+      end
+
+      migrate_data(table_name)
```

Comment:
I would also add a CHECK constraint that will require certain fields depending on `created_by` enum field.

Reply:
Gotcha, that makes sense.  I'm guessing that enum value would be `created_by` = `identity` in this case?

Reply:
Yeah, and `actor` if identity_id is null.

Reply:
As I tried to work this CHECK out I realized that all of the Enum values aren't the same for every table.  Would you prefer a simple check that makes sure the `created_by` field isn't `identity` if the `created_by_identity_id` is NULL or would you prefer an individual customized CHECK per table?

Reply:
Unfortunately, I don't think that will work for a couple of reasons.  The biggest being that we can't reference columns that don't exist on the table.  The second reason (at least in the local Dev DB, not sure if prod and staging are different) is that which ever value is in `created_by` doesn't guarantee that all of the other `created_by_<something>_id` fields will be NULL.

The second reason I need to dig in to a little bit more to see if that's just a side effect of how we seed our Dev DBs or if it actually ends up in that state in staging/prod.